sdk/core/Azure.Core/src/Internal/AzureCorePipelineProcessor.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Azure.Core.Pipeline
+{
+    internal struct AzureCorePipelineProcessor : IReadOnlyList<PipelinePolicy>
```

Comment:
Add a comment

Reply:
Why it is a struct?

Reply:
My thinking was that if it only holds two pointers we might get a small perf win by keeping it on the stack instead of the heap, but I did not measure to validate my hypothesis.  Do you have a concern?  If so, could you articulate the problem?  It's also internal so we can change it at any time.